### ISSUE TITLE
Tune app startup

### ISSUE THREAD

**Issue by sonora** [osmandapp/Osmand#2926]
I receive a lot of feedback lately from users who are annoyed that after app startup it takes a long time looking at a black , later gray screen before the map comes up. (Position marker is actually long present on the gray screen already). Some time ago it was the other way round: Map showed quicker, the position marker took longer, people seem to have preferred that.

Not sure we can do much about it, but UX is a big portion of the app and how quick the map shows affects usability in many cases, so I wanted to document this request.

**Comment by stephan75** [comment]
What about introducing a time measurement in osmands dev plugin to display the time needed from app start till map display finish?

**Comment by sonora** [comment]
10 seconds of black screen, followed by another 15+ seconds of gray screen, from what I measure on most devices.

After that the map shows, and that's what most people want. (Although the app is not completely "ready" yet for all purposes like QuickSearch may not be able to display results yet, and if a navigation is started, the TTS engine may take extremely long (up to 2 minutes, as reported elsewhere) until producing sound, indicating that something else is still going on. (Waiting only a few more seconds may produce TTS sound immediately.)

 But all that are side notes, what I want to report here is that most people miss that they cannot quickly take a glance at the map without having to about a half minute or so.

**Comment by polarbearing** [comment]
Fully agree with the issue, I just did not dare to report it yet because I was afraid my phone might be too old ;-)
Not sure if it is related, but it sometimes takes quite a long time after a GPS fix to show the current road in non-navigation mode.

**Comment by vshcherb** [comment]
Agree, it is a bit annoying. On my device it takes about 9-11 seconds to show the map. Detailed analysis what's loaded is available in the logs. So any questions/suggestions are welcome. I guess this issue is not about introducing progress bar on the map screen, though it is also possible.

**Comment by sonora** [comment]
Introducing a progress bar could help psychologically ... which is a valid suggestion. Still better would be we analyze in what order we initialize what, and see if there is any way to get back closer to where we were with previous releases, where the maps seems to have shown a tad quicker. Can you share a copy of the log so we can take a look?

**Comment by utack** [comment]
Stupid question:  
I think at one point you planned multicore map rendering, how is the status on that one?  
Maybe you can shift this feature back a bit, but swat it with a proper solution that improves the entire app.

**Comment by vshcherb** [comment]
If I share my timing it won't mean anything cause mine is twice faster and I already was improving by 2.1 , so this code is nicely structured and I didn't find anything to improve https://github.com/osmandapp/Osmand/blob/master/OsmAnd/src/net/osmand/plus/AppInitializer.java#L483.

**Comment by stephan75** [comment]
Is there any profiling tool available for that part of osmand source code?

**Comment by vshcherb** [comment]
LogCat or any other logging program can help you to capture logs (android sdk has adb tool)
